lib/streamlit/elements/radio.py

```diff
@@ -67,7 +73,7 @@ def radio(
         *,  # keyword-only args:
         disabled: bool = False,
         horizontal: bool = False,
-    ) -> Any:
+    ) -> Optional[V_co]:
```

Comment:
This feels a bit weird to me given that we have that the only case where `None` can be returned from `st.radio` (and also `st.selectbox`) is when `options` is empty. It took a bit of thought before I realized why this return type is the correct one rather than it simply being `V_co`.

I think there's not much we can actually do about this, though, since I highly doubt there's any way to express a condition on the length of `options` in Python's (or any popular/actually-used language's) type annotations, so I think we just have to accept this assuming we want `st.radio` and `st.selectbox` to be accurately typed.

Reply:
I'll give you a very non-serious proposal for how to avoid this return type. It _is_ possible, it just isn't very suited for a library that tries to be forgiving to relatively inexperienced coders, as it introduces some quite non-pythonic concepts.

The proposal is based on a (quite good!) blog post on avoiding validation in code: https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/. It's worth a read in full.

Anyway, the author proposes a non-empty list type. Imagine implementing the same concept in streamlit, and, instead of requiring an `OptionSequence[V_co]`, you require a `NonEmptySequence[V_co]`. This would mean that the function would unconditionally return `V_co`.

From my personal point of view, I would actually appreciate this solution. I don't have anything against being forced to parse my input sequence early in order to avoid failures later, but I understand that this extra indirection is not everyone's cup of tea, and it definitely makes the learning curve for your API harder.

So, TL;DR: @vdonato is probably right.

Reply:
Oh neat. Introducing a new type to solve this problem is something that I didn't consider. I should really read more into type-driven design as I've been meaning to for quite some time now but have never gotten around to doing it. I'll give the blog post a more thorough read when I have some more spare time 🙂 

I think the majority opinion internally is that many of us would also be quite happy being forced to do extra work upfront to avoid running into more unexpected issues later, but doing this runs counter to some of our user-friendliness goals.